client/src/elm/Pages/WellChild/ProgressReport/View.elm

```diff
@@ -2675,13 +2701,75 @@ viewFillTheBlanksPane language currentDate zscores child db groupNutritionMeasur
                                 )
                                 set.height
                         )
-                        allValuesSets
+                        allValuesSetFromNutrition
                 )
                 maybeAgeInDays
                 |> Maybe.withDefault []
-                |> distributeByAgeInMonths child
+                |> distributeByAgeInMonthsWithDate child
+
+        heightsByAgeInMonthsWithDateFromNCDA =
+            List.filterMap
+                (\( date, set ) ->
+                    Maybe.map
+                        (\stuntingLevel ->
+                            case stuntingLevel of
+                                LevelRed ->
+                                    ( date, NCDACellValueT )
+
+                                LevelYellow ->
+                                    ( date, NCDACellValueH )
+
+                                LevelGreen ->
+                                    ( date, NCDACellValueC )
+                        )
+                        set.stuntingLevel
+                )
+                ncdaValuesSet
+                |> distributeByAgeInMonthsWithDate child
 
         weightsByAgeInMonths =
+            case ( weightsByAgeInMonthsWithDateFromNutrition, weightsByAgeInMonthsWithDateFromNCDA ) of
+                ( Just fromNutrition, Just fromNCDA ) ->
+                    Dict.merge
+                        (\key value -> Dict.insert key (Tuple.second value))
+                        -- In case we got both values for month, we give preference to
```

Comment:
May we reduce code redundancy with https://github.com/TIP-Global-Health/eheza-app/pull/905/files#diff-53236e6f07f593ca22091310ef666c976e601040442e42239c653b9bc73ef6a2R2673 ?

Reply:
@AronNovak done